src/index.ts

```diff
@@ -15,24 +15,27 @@ const PARENT_DIRECTORY = /^(\/?\.\.)+/;
 const ESCAPING_BACKSLASHES = /\\(?=[()[\]{}!*+?@|])/g;
 const BACKSLASHES = /\\/g;
 
+// The `Array.isArray` type guard doesn't work for readonly arrays.
+const isReadonlyArray: (arg: unknown) => arg is readonly unknown[] = Array.isArray;
```

Comment:
you forgot to use this on line 181 👀 i wonder if we should move this to utils?

Reply:
<img width="752" height="126" alt="image" src="https://github.com/user-attachments/assets/e091b5a1-1ff6-4200-90dc-7d13a2e7004b" />
it types `patterns` as `any[]`. but it doesn't do any harm right now since we aren't accessing the array elements. not sure what to do though

Reply:
Yeah I think this is fine in its current state! It just happens to be that this particular usage doesn't need the readonly array type guard